wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketServerURLParser.java

```diff
@@ -114,9 +114,9 @@ private boolean isApiRequestRelevant(String repositoryUrl) {
               new BitbucketServerOAuthAuthenticator("", "", "", ""),
               oAuthAPI,
               "");
-      // If the token request catches the unauthorised error, it means that the provided url
+      // If the user request catches the unauthorised error, it means that the provided url
       // belongs to Bitbucket.
-      bitbucketServerApiClient.getPersonalAccessToken("", 0L);
+      bitbucketServerApiClient.getUser(null);
```

Comment:
why do we pass null as an argument?

Reply:
If `null` is passed the client fetches OAuth token:
https://github.com/eclipse-che/che-server/blob/1657ebbaa9582bda79df30dc31f8003fadcd0654/wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/server/BitbucketServerApiClient.java#L30-L38